reana_server/reana_admin/main.py

```diff
@@ -639,3 +640,61 @@ def set_default_quota_limit(ctx):
             resource_type=resource.type_.name,
             limit=DEFAULT_QUOTA_LIMITS.get(resource.type_.name),
         )
+
+
+@reana_admin.command("consume-queue")
+@click.option(
+    "--queue-name",
+    "-q",
+    required=True,
+    type=str,
+    help="Name of the queue that will be consumed, e.g workflow-submissions",
+)
+@click.option(
+    "--key",
+    "-k",
+    type=str,
+    help="Key that will be used to get the id value from the message body, e.g workflow_name_or_id",
+)
+@click.option(
+    "--messages-to-delete",
+    "-m",
+    multiple=True,
+    help="List of message ids to remove from the queue, e.g UUID of a workflow",
+)
+@click.option(
+    "-i",
+    "--interactive",
+    is_flag=True,
+    default=False,
+    help="Manually decide which messages to remove from the queue.",
+)
+def consume_queue(
+    queue_name: str,
+    key: Optional[str],
+    messages_to_delete: List[str],
+    interactive: bool,
+):
+    """Start consuming specified queue and remove selected messages."""
+    from reana_server.reana_admin.consumer import MessageConsumer
+
+    if key and interactive:
```

Comment:
I think it might be useful to be able to specify both at the same time. In this way, the administrators can filter the messages based on some conditions (e.g. `user` is `xyz`), but they can still decide manually whether to remove each of those message from the queue. The other messages are simply requeued.

Reply:
Can you describe more what it will look like from the administrator's perspective? Like what the input and output of the script will look like.

Reply:
For messages that do not match the given filters, we can either display nothing or show the received message along with `Message is ignored and re-queued`. If a message matches the given filters, we ask the user whether he wants to drop it or not.

In summary:
- if only `-k` (and `--values-to-delete`) is specified, the script drops all the messages that match the given filter
- if only `-i` is specified, the user decides what to do with each message
- if both `-i` and `-k` are specified, the user decides what to do with each message *that matches the given filter*

Do you think this is a good approach?

Reply:
I think this approach makes sense. I will change it.

Reply:
Done.